kn/fotos/entities.py

```diff
@@ -269,42 +282,26 @@ def update_metadata(self):
             if 'DateTimeOriginal' in exif:
                 self.created = datetime.datetime.strptime(exif['DateTimeOriginal'], '%Y:%m:%d %H:%M:%S')
 
+        if self.size is None:
+            self.size = img.size
+
+        return True
 
     def _cache(self, cache):
         if cache == 'full':
-            return {}
+            return
         source = self.original_path
         target = self.get_cache_path(cache)
         target_dir = os.path.dirname(target)
         if not os.path.exists(target_dir):
             os.makedirs(target_dir)
-        if cache == 'thumb':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '200x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'thumb2x':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '400x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'large':
-            subprocess.check_call(['convert',
-                             source,
-                             '-resize', '850x',
-                             '-rotate', str(self.rotation),
-                             target])
-        elif cache == 'large2x':
-            subprocess.check_call(['convert',
-                             source,
-                             '-quality', '90',
-                             '-resize', '1700x',
-                             '-rotate', str(self.rotation),
-                             target])
-        # No worries: Image.open is lazy and will only read headers
-        return {'size': Image.open(target).size}
+
```

Comment:
In ieder geval `large2x` iets lager zetten.  De rest kan voorlopig nog zonder argument, wat mij betreft.  Als de foto's ongemakkelijk groot blijken (waarschijnlijk alleen `thumb2x`), dan kunnen we het nog aanpassen.